src/PhpDoc/ResolvedPhpDocBlock.php

```diff
@@ -40,10 +40,10 @@ class ResolvedPhpDocBlock
 	 * @param \PHPStan\PhpDoc\Tag\ReturnTag|null $returnTag
 	 */
 	public function __construct(
-		array $varTags,
-		array $methodTags,
-		array $propertyTags,
-		array $paramTags,
+		array $varTags = [],
+		array $methodTags = [],
+		array $propertyTags = [],
+		array $paramTags = [],
```

Comment:
I really don't like optional parameters :)

Reply:
That'd be much nicer! You can even make the constructor private and make two `create*` static methods.